### PR TITLE
Add calico configuration support

### PR DESCRIPTION
**Pull request by mainred**
Add calico configurations into terraform and make several small fixes. 

fix: [DCOS-62550](https://jira.mesosphere.com/browse/DCOS-62550) integration calico configuration into terraform

### CODE REVIEW

**Review by wavesoft — DISMISSED** [review]
LGTM! 👍

**Review by fatz — CHANGES_REQUESTED** [review]
the minimum core version should be `version = "~> 0.2.12"` as previous releases do not support the new vars

File: main.tf [hunk 1]
```diff
@@ -64,7 +64,7 @@
 
 module "dcos-core" {
   source  = "dcos-terraform/dcos-core/template"
-  version = "~> 0.2.11"
+  version = "~> 0.2.12"
```
Comment (mainred) [inline main.tf:67]: `dcos-terraform/dcos-core/template` will cut a new release to include calico configurations

File: main.tf [hunk 2]
```diff
@@ -90,6 +90,14 @@ module "dcos-core" {
   dcos_ca_certificate_chain_path               = "${var.dcos_ca_certificate_chain_path}"
   dcos_ca_certificate_key_path                 = "${var.dcos_ca_certificate_key_path}"
   dcos_ca_certificate_path                     = "${var.dcos_ca_certificate_path}"
+  dcos_calico_network_cidr                     = "${var.dcos_calico_network_cidr}"
+  dcos_calico_vxlan_enabled                    = "${var.dcos_calico_vxlan_enabled}"
+  dcos_calico_vxlan_port                       = "${var.dcos_calico_vxlan_port}"
+  dcos_calico_ipinip_mtu                       = "${var.dcos_calico_ipinip_mtu}"
+  dcos_calico_vxlan_mtu                        = "${var.dcos_calico_vxlan_mtu}"
+  dcos_calico_vxlan_port                       = "${var.dcos_calico_vxlan_port}"
```
Comment (fatz) [inline main.tf]: `dcos_calico_vxlan_port` is duplicated

**Review by fatz — CHANGES_REQUESTED** [review]
please have a look at my comment. The var gets forwarded twice.

I think it will run anyways but lets keep this clean

File: main.tf (hunk 2 above)
Comment (mainred) [reply main.tf]: Done, thanks. 
Can we test these changes in our local env?

Comment (fatz) [reply main.tf]: yes but you have to use our forked terraform build which supports `Terraformfile` 

https://github.com/fatz/terraform/releases/tag/v0.11.14-mesosphere the feature is documented here https://github.com/hashicorp/terraform/pull/20229

With that you can overwrite module sources.

This is also used by Jenkins. I've used this `Terraformfile` to test your changes

```json
{
  "dcos-terraform/dcos/aws": {
    "source": "git::https://github.com/mainred/terraform-aws-dcos.git?ref=add-calico-configurations"
  },
  "dcos-terraform/dcos-install-remote-exec/null": {
    "source": "git::https://github.com/mainred/terraform-null-dcos-install-remote-exec.git?ref=add-calico-configurations"
  },
  "dcos-terraform/dcos-core/template": {
    "source": "git::https://github.com/mainred/terraform-template-dcos-core.git?ref=add-calico-configurations"
  }
}
```

**Review by fatz — APPROVED** [review]

**Pull request merged**